### PR TITLE
feat: Framework Splitting and Bumpenvs

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Ticket
[MD-410](https://hpe-aiatscale.atlassian.net/browse/MD-410)

## Description
Revert https://github.com/determined-ai/determined/pull/9405
Bumpenvs for NGC+ images
Various fixes for slurm builds and runs
Docs table changes
Drop `--cuda` and `--cpu` from hpc launcher in CI
Adds afw efs test update by @azhou-determined 

## Test Plan
CI passes

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[MD-410]: https://hpe-aiatscale.atlassian.net/browse/MD-410?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by keita-determined — APPROVED** [review]

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Review by loksonarius — APPROVED** [review]

File: e2e_tests/tests/config.py
Comment (MikhailKardash) [reply e2e_tests/tests/config.py]: Oops, this was me working around a bad circleci config in `test_e2e_gke`. I'll just fix that config instead.

**Review by AmanuelAaron — APPROVED** [review]

**Review by azhou-determined — APPROVED** [review]
lgtm, assuming all the tests pass 🙃

**Review by hamidzr — APPROVED** [review]
changes under `master/` lgtm

File: .circleci/scripts/pull_image_daemonset.yaml
Comment (davidfluck-hpe) [inline .circleci/scripts/pull_image_daemonset.yaml:1]: Why are we using a daemonset to pull a Docker image onto each node?

Comment (MikhailKardash) [reply .circleci/scripts/pull_image_daemonset.yaml:1]: The NGC+ images that we are moving to take a significant time to pull, which causes some unit tests to hang because they end up pulling these images at runtime. Using a daemonset to pull them first seems easier and more reliable.

Comment (davidfluck-hpe) [reply .circleci/scripts/pull_image_daemonset.yaml:1]: Are there Docker credentials in the build process (in CircleCI) somewhere that could be used to pull the image more explicitly instead? The daemonset technically works, but it's being used more for its side effect of pulling images than for running specific workloads on all Kubernetes nodes.

Comment (MikhailKardash) [reply .circleci/scripts/pull_image_daemonset.yaml:1]: Well, these images are public to begin with, so credentials wouldn't be necessary. We just need some mechanism to make sure that all pods pull this image.

Comment (davidfluck-hpe) [reply .circleci/scripts/pull_image_daemonset.yaml:1]: If you specify it on whatever pod needs it, Kubernetes will fetch the image a first time. Subsequent workloads that use `imagePullPolicy: IfNotPresent` will then use the node-local version. Does that not work? Also, are the subsequent unit tests using Kubernetes workloads specifically? Or just interacting directly with Docker?

Comment (MikhailKardash) [reply .circleci/scripts/pull_image_daemonset.yaml:1]: The subsequent unit tests are just interacting with docker. I'm not sure if it's easier to enforce using specific pods rather than just making sure all pods have this image installed.

Comment (davidfluck-hpe) [reply .circleci/scripts/pull_image_daemonset.yaml:1]: I'm still a bit confused. How are the unit tests running docker? Are the tests themselves running from inside pods and also interacting with docker there? (I.e. a "docker in docker" situation.) Or is something running `docker <blah>` from outside of Kubernetes?

Comment (davidfluck-hpe) [reply .circleci/scripts/pull_image_daemonset.yaml:1]: Because if the tests are running inside K8s pods, then you shouldn't need the daemonset at all. The first pod to run the test that requires this image will pull it, then subsequent pods, if using `imagePullPolicy: IfNotPresent` should use the existing image that K8s already has.

Comment (MikhailKardash) [reply .circleci/scripts/pull_image_daemonset.yaml:1]: We use a `run-e2e-tests` command in circleci. that doesn't use k8s at all, but rather `pytest` with a remote master target. All it does is schedule jobs on a remote cluster, but it doesn't know whether it's a k8s cluster or a normal determined cluster.
When the tests are requested, the remote cluster schedules it's own pods.

The particular tests that timeout due to docker pulls are a combination of two factors:

1. The default image in the master config is not updated because this master is not being created/restarted.
2. The test itself schedules another experiment, waits for that experiment to finish, then exits. Since this launches a new task, a new pod gets allocated and has to do a pull. The test itself has to wait for that pull to finish.

Comment (davidfluck-hpe) [reply .circleci/scripts/pull_image_daemonset.yaml:1]: Thanks for the context! I'm still a bit nervous, but I don't want to hold this up more without being able to devote more time to thinking about it, so I'll approve this.

**Review by davidfluck-hpe — APPROVED** [review]

**Pull request merged**
